packages/patterns/src/keys/compareKeys.js

```diff
@@ -147,6 +147,30 @@ export const compareKeys = (left, right) => {
       // @ts-expect-error narrowed
       return compareRank(left.length, right.length);
     }
+    case 'byteArray': {
+      // @ts-expect-error narrowed
+      const leftArray = new Uint8Array(left.slice(0));
+      // @ts-expect-error narrowed
+      const rightArray = new Uint8Array(right.slice(0));
+      // @ts-expect-error narrowed
+      const byteLen = Math.min(left.byteLength, right.byteLength);
+      for (let i = 0; i < byteLen; i += 1) {
+        const leftByte = leftArray[i];
+        const rightByte = rightArray[i];
+        if (leftByte < rightByte) {
+          return -1;
+        }
+        if (leftByte > rightByte) {
+          return 1;
+        }
+      }
```

Comment:
Small optimization: copy only when necessary.
```suggestion
      // Account for gaps in the @endo/immutable-arraybuffer shim.
      const leftArray =
        Object.getPrototypeOf(left) === ArrayBuffer.prototype
          // @ts-expect-error narrowed
          ? new Uint8Array(left)
          // @ts-expect-error narrowed
          : new Uint8Array(left.slice(0));
      const rightArray =
        Object.getPrototypeOf(right) === ArrayBuffer.prototype
          // @ts-expect-error narrowed
          ? new Uint8Array(right)
          // @ts-expect-error narrowed
          : new Uint8Array(right.slice(0));
      // @ts-expect-error narrowed
      const byteLen = Math.min(left.byteLength, right.byteLength);
      for (let i = 0; i < byteLen; i += 1) {
        const leftByte = leftArray[i];
        const rightByte = rightArray[i];
        if (leftByte < rightByte) {
          return -1;
        }
        if (leftByte > rightByte) {
          return 1;
        }
      }
```
And also, why not implement shortlex ordering for ByteArrays from the start?